src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/MinMax.cs

```diff
@@ -35,39 +35,50 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             Contracts.AssertValue(errors);
             Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
 
-            nodeToCoercedTypeMap = new Dictionary<TexlNode, DType>();
+            nodeToCoercedTypeMap = new Dictionary<TexlNode, DType>();            
             var fArgsValid = true;
             returnType = argTypes[0];
 
-            // If there is mixing of Date and DateTime, coerce Date to DateTime
-            if (Array.TrueForAll(argTypes, element => element.Kind == DKind.Date || element.Kind == DKind.DateTime) && !Array.TrueForAll(argTypes, element => element.Kind == DKind.Date))
+            // When PowerFxV1CompatibilityRules is enabled, return type is always argTypes[0]
+            if (context.Features.PowerFxV1CompatibilityRules)
             {
                 for (var i = 0; i < argTypes.Length; i++)
```

Comment:
>0 [](http://example.com/codeflow?start=29&length=1)

nit:  start at  1? 
is this comparing 0 to itself.

Reply:
I see - starting at 0 is key because it will catch if arg0 is not a valid type (like record).